### PR TITLE
[core] Added support for multiple containers in a pod for logs

### PR DESCRIPTION
**Pull request by radtriste**
Issue #296: Problem when getting logs of a pod with many containers -
Suggestion for API improvement

### CODE REVIEW

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java [hunk 1]
```diff
@@ -372,6 +444,38 @@ public boolean deletePods(String key, String value) {
...
+	}
+	
+	public Container getContainerByName(Pod pod, String containerName) {
+		return getAllContainers(pod).stream()
+			.filter(c -> c.getName().equals(containerName))
+			.findFirst()
+			.orElseThrow(() -> new RuntimeException("Cannot find container with name " + containerName + " in pod " + pod.getMetadata().getName()));
+	}
+
+	private <R> Map<String, R> retrieveFromPodContainer(Pod pod, Function<Container, R> containerRetriever) {
```
Comment (Crumby) [inline core/src/main/java/cz/xtf/core/openshift/OpenShift.java]: Please make a method name plural so it's consistent with return type (`retrieveFromPodContainer`). This way it seems it should be from one container.

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java [hunk 2]
```diff
@@ -372,6 +444,38 @@ public boolean deletePods(String key, String value) {
...
+	 * 
+	 * @param pod Pod to retrieve from
+	 * @return One random container from the pod
+	 */
+	public Container getAnyContainer(Pod pod) {
+		List<Container> containers = getAllContainers(pod);
+		return containers.get(new Random().nextInt(containers.size()));
+	}
+	
+	public Container getContainerByName(Pod pod, String containerName) {
```
Comment (Crumby) [inline core/src/main/java/cz/xtf/core/openshift/OpenShift.java]: Make it just `getContainer` so it's consistent with other getResource methods.

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java [hunk 3]
```diff
@@ -292,21 +297,88 @@ public String getPodLog(String deploymentConfigName) {
...
+	public Map<String, String> getAllContainerLogs(Pod pod) {
+		return retrieveFromPodContainer(pod, container -> this.getPodLog(pod, container));
 	}
 
 	public Reader getPodLogReader(Pod pod) {
-		return pods().withName(pod.getMetadata().getName()).getLogReader();
+		return getPodLogReader(pod, getAnyContainer(pod));
+	}
+
+	public Reader getPodLogReader(Pod pod, Container container) {
```
Comment (Crumby) [inline core/src/main/java/cz/xtf/core/openshift/OpenShift.java]: Please reorganize the method so the final its top down from the one with the simplest parameters to the actual implementation:
```
getPodLogReader(Pod pod)
getPodLogReader(Pod pod, String containerName)
getPodLogReader(Pod pod, Container container)
```

Applies also for others updated and created methods.

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java (hunk 1 above)
Comment (radtriste) [reply core/src/main/java/cz/xtf/core/openshift/OpenShift.java]: updated

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java (hunk 2 above)
Comment (radtriste) [reply core/src/main/java/cz/xtf/core/openshift/OpenShift.java]: updated

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java (hunk 3 above)
Comment (radtriste) [reply core/src/main/java/cz/xtf/core/openshift/OpenShift.java]: updated

**Pull request merged**